tests/test_tc101.py

```diff
@@ -113,8 +114,38 @@ def foo(self) -> 'X':
         '''),
         set(),
     ),
+    # Make sure we didn't introduce any regressions while solving #167
+    # since we started to treat the RHS sort of like an annotation for
+    # some of the use-cases
+    (
+        textwrap.dedent('''
+        from __future__ import annotations
+        if TYPE_CHECKING:
+            from foo import Foo
+
+        x: TypeAlias = 'Foo'
+        '''),
+        set(),
+    ),
 ]
 
+if sys.version_info >= (3, 12):
+    examples.append(
+        (
+            # Make sure we didn't introduce any regressions while solving #167
+            # using new type alias syntax
+            # TODO: Make sure we actually need to wrap Foo if we use future
```

Comment:
My assumption would be yes, since it's the right hand side of the assignment. Did you find out?

Reply:
It would also be my assumption, but it's not an assignment in the AST, it is its own node, so they could treat it differently and it would make some sense in the context of all annotations being treated as a ForwardRef, it makes less sense with `TypeAlias`, since that could be any number of things at runtime, but with the new notation they know statically at parse-time that the RHS is a type alias.

I haven't gotten around to installing python 3.12 locally yet, so no.

Reply:
I just tried it and it turns out it has no runtime effect regardless of the future import, so it looks like you never need to wrap a type alias with the new syntax ever. So it probably makes more sense to add a special error code like TC006, to tell people that they should never wrap an alias using the new syntax.

We will still need to treat the old explicit style of `x: TypeAlias = ...` separately though, since those need to be wrapped/unwrapped regardless of whether we opted into TC1xx or TC2xx. For unwrapping we have an appropriate error for both styles, but not for the reverse case.

Reply:
I think I will just go with adding:

TC007: Type alias '{alias}' needs to be made into a string literal
TC008: Type alias '{alias}' does not need to be a string literal

The latter makes sense for both kinds of type aliases, the former only for the old explicit style.